### PR TITLE
Create a hierarchical table for syllables and motifs

### PR DESCRIPTION
**Pull request by weiglszonja**
The `MotifInterface` is modified to add the timings of syllables along with the motifs.
The motifs are added to the trials table and is a `HierarchicalBehavioralTable` where the
lowest hierarchical level is the level of syllables. The syllables are added to the file as `TimeIntervals` object.

### CODE REVIEW

**Comment by weiglszonja** [comment]
Figuring out this error:


`hdmf.build.errors.ReferenceTargetNotBuiltError: next_tier (root/processing/behavior/Motifs/next_tier): Could not find already-built Builder for TimeIntervals 'Syllables' in BuildManager`

**Comment by weiglszonja** [comment]
@CodyCBakerPhD Figured out that error, problem is I'm only adding the motifs table to the processing module, I have to add the syllables too to make the reference work!

```
# Add the hierarchical table to the processing module
behavior_module.add(syllables)
behavior_module.add(motifs_table)
```

**Comment by CodyCBakerPhD** [comment]
Interesting, you can't have a hierarchical reference to the global TrialsTable? I suppose that's fine but I'd be curious as to why it's made that way

File: src/fee_lab_to_nwb/happ_ecephys/happmotifinterface.py [hunk 1]
```diff
@@ -57,23 +61,75 @@ def get_synchronized_motif_timestamps(self):
 
         return motif_timestamps
 
+    def create_hierarchical_table_from_motif_timestamps(self, motif_timestamps: ArrayType):
+        """Create a hierarchical table from the timings of motifs.
+        The lowest hierarchical level is the level of syllables. The number of
+        syllables within a motif is assumed to be fixed per experiment. The latency
+        of each syllable is fixed for now, but should be modified in the future given
+        more information about the mapping between motifs and syllables."""
+        syllables = TimeIntervals(name="Syllables", description="desc")
```
Comment (weiglszonja) [inline src/fee_lab_to_nwb/happ_ecephys/happmotifinterface.py:70]: Ah I see what you mean, I think that should be possible, but currently I'm not doing it that way, just create a `TimeIntervals` object for the syllables. But I'll try it that way!

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**